loader/src/main/kotlin/net/weavemc/loader/WeaveLoader.kt

```diff
@@ -53,15 +47,25 @@ class WeaveLoader(
     }
 
     private fun finalize() {
+        println("[Weave] Finalizing Weave")
         retrieveMods().forEach { it.registerAsMod() }
+        println("[Weave] Verifying dependencies")
         verifyDependencies()
+        println("[Weave] Populating mixin modifiers")
         populateMixinModifiers()
 
+        println("[Weave] calling preInit() for mods")
         // TODO remove
         // Invoke preInit() once everything is done.
         mods.forEach { weaveMod ->
             weaveMod.config.entryPoints.forEach { entrypoint ->
-                instantiate<ModInitializer>(entrypoint).preInit(instrumentation)
+                runCatching {
+                    @Suppress("DEPRECATION")
+                    instantiate<ModInitializer>(entrypoint).preInit(instrumentation)
+                }.onFailure {
+                    it.printStackTrace()
+                    println("Failed to instantiate $entrypoint#preInit")
```

Comment:
Not a big thing but the exception can be thrown from preInit too.

Reply:
This is just to print the exception, instead of letting it propagate. Catching from preinit is intended. Should I change it to have two different handlers?

Reply:
You could do something like `runCatching { instantiate }.onFailure { ... }.onSuccess { (try-catch calling it.init()) }`

Reply:
I meant the error message conveys that it's caused from instantiating.